### PR TITLE
OCPBUGS-14012: Allow created NICs to inherit tags

### PR DESCRIPTION
**Pull request by odvarkadaniel**
This PR enables newly created NICs to inherit tags from a machine.

### CODE REVIEW

**Comment by odvarkadaniel** [comment]
This PR might be solving the issue already: https://github.com/openshift/machine-api-provider-azure/pull/45
I don't think this 2 line addition is needed when we already have the aformentioned PR merged. It would just need to get back-ported all the way to 4.11 to fix the bug `OCPBUGS-14012`. What do you think, @JoelSpeed?

**Comment by JoelSpeed** [comment]
Looking at #45, it doesn't look to me like it is tagging the NICs, but is only tagging the VMs right? We need the NICs tagged too so I think to resolve the real issue, we need to have both PRs merged in, let me know if you think that's not right

**Comment by odvarkadaniel** [comment]
> Looking at #45, it doesn't look to me like it is tagging the NICs, but is only tagging the VMs right? We need the NICs tagged too so I think to resolve the real issue, we need to have both PRs merged in, let me know if you think that's not right

I think so now as well. As you are saying, this PR and #45 combined is a valid fix IMO. I will create a manual back-port to 4.13 once this can get merged and etc.

File: pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ type Spec struct {
 	SecurityGroupName             string
 	ApplicationSecurityGroupNames []string
 	AcceleratedNetworking         bool
+	Tags                          map[string]*string
```
Comment (JoelSpeed) [inline pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go:52]: Does this need to be plumbed through to the network spec? This I think is our internal version that then needs to be converted in the `CreateOrUpdate` function

Comment (odvarkadaniel) [reply pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go:52]: I think that it needs to be there as the tags for NICs are inherited by VMs like [here](https://github.com/openshift/machine-api-provider-azure/blob/main/pkg/cloud/azure/services/virtualmachines/virtualmachines.go#L115) - we need to add not only `Name`, but `Tags` as well. Therefore we need to introduce `Tags` to our internal representatino as well, I think.

Comment (JoelSpeed) [reply pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go:52]: Does https://github.com/openshift/machine-api-provider-azure/blob/eda92b6117b0b46deeaa78bd120c20176f2150b3/pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go#L303C2-L303C2 mean that this isn't actually needed? WDYT? This is what I meant earlier, there should be tags being passed to the NIC create call, except it looked like there wasn't. Now I see they are taking them from the scope, so this field you're adding appears to be unused right?

Comment (odvarkadaniel) [reply pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go:52]: No, the line you are mentioning is for sure needed - I think that this creates the actual network interface in azure.
Yes, that's why I initially thought that this PR is not needed at all, but I think we need the new field [here](https://github.com/openshift/machine-api-provider-azure/blob/eda92b6117b0b46deeaa78bd120c20176f2150b3/pkg/cloud/azure/actuators/machine/reconciler.go#L562) as the creation is called [here](https://github.com/openshift/machine-api-provider-azure/blob/eda92b6117b0b46deeaa78bd120c20176f2150b3/pkg/cloud/azure/actuators/machine/reconciler.go#L639) - with the spec that contains the correct tags.

Comment (odvarkadaniel) [reply pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go:52]: Update: Tested this on a live cluster and this PR doesn't change anything. I think I can close this as the PR #45 already fixes that.
Edited an existing MachineSet by adding a tag and a new NIC got created with our tag.

**Pull request closed without merging**